backend/app/services/rag/retrieval_service.py

```
@@ -39,6 +49,420 @@ def __init__(self, storage_backend: Optional[BaseStorageBackend] = None):
+            return 0
+
+        document_query = db.query(
+            func.coalesce(func.sum(SubtaskContext.text_length), 0)
+        )
+        document_query = document_query.select_from(KnowledgeDocument).join(
+            SubtaskContext,
+            KnowledgeDocument.attachment_id == SubtaskContext.id,
+        )
+        document_query = document_query.filter(
+            KnowledgeDocument.kind_id.in_(knowledge_base_ids),
+            KnowledgeDocument.is_active == True,
```

Ruff (0.15.7)

[error] Avoid equality comparisons to `True`; use `KnowledgeDocument.is_active:` for truth checks

Replace with `KnowledgeDocument.is_active`

(E712)